aspnetcore/mvc/extensibility/app-parts.md

```diff
@@ -0,0 +1,122 @@
+var part = new AssemblyPart(assembly);
+services
+    .AddMvc()
+    .ConfigureApplicationPartManager(apm => p.ApplicationParts.Add(part));
+```
+
+The main use case for application parts is to allow you to configure your app to discover MVC features from another assembly. For instance, by default MVC will search the dependency tree and find controllers (even in other assemblies). To load an arbitrary assembly (for instance, from a plugin that isn't referenced at compile time), you can use an application part. Without application parts, the only way to achieve this would be to replace and rewrite controller discovery and creation - using application parts greatly simplifies this process.
+
+Note that you can also use application parts to *avoid* looking for controllers in a particular assembly or location. By modifying the `ApplicationParts` collection of the `ApplicationPartManager`, you can control which parts (or assemblies) are available to the app. The order of the entries in the `ApplicationParts` collection is not important. It is important, though, to ensure you have configured your application parts before you try to use them for features. In `ConfigureServices`, be sure to configure the application part manager completely before using anything that requires services these parts may have added.
+
+If you have an assembly in your MVC project's dependency tree that has controllers in it you do not want to be used by the app, remove it from the `ApplicationPartManager`:
+
+<!-- literal_block {"ids": [], "linenos": true, "xml:space": "preserve", "language": "csharp"} -->
+
+```csharp
+services.AddMvc()
+    .ConfigureApplicationPartManager(p =>
+    {
+        var dependentLibrary = p.ApplicationParts
+            .FirstOrDefault(part => part.Name == "DependentLibrary");
+        if (dependentLibrary != null)
+        {
+           p.ApplicationParts.Remove(dependentLibrary);
+        }
+    })
+```
+
+In addition to your project's assembly and its dependent assemblies, the `ApplicationPartManager` will include parts for `Microsoft.AspNetCore.Mvc.TagHelpers` and `Microsoft.AspNetCore.Mvc.Razor` by default.
+
+## Application Feature Providers
+
+Application Feature Providers examine application parts and provide features for those parts. There are built-in feature providers for the following MVC features:
+
+- [Controllers](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.controllers.controllerfeatureprovider)
+- [Metadata Reference](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.razor.compilation.metadatareferencefeatureprovider)
+- [Tag Helpers](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.razor.taghelpers.taghelperfeatureprovider)
+- [View Components](https://docs.microsoft.com/aspnet/core/api/microsoft.aspnetcore.mvc.viewcomponents.viewcomponentfeatureprovider)
+
+> [!NOTE]
+> Views do not use feature providers but will do so in a future release.
```

Comment:
@ardalis  roger that. Might be OK to say it will be used in the future, need to check with @danroth27 .